sdk/schemaregistry/azure-schemaregistry-avroserializer/azure/schemaregistry/serializer/avroserializer/_schema_registry_avro_serializer.py

```diff
@@ -85,12 +89,13 @@ def _get_schema_id(self, schema_name, schema, **kwargs):
         try:
             return self._schema_to_id[schema_str]
         except KeyError:
-            schema_id = self._schema_registry_client.register_schema(
-                self._schema_group,
-                schema_name,
-                "Avro",
-                schema_str,
-                **kwargs
+            schema_id_func = (
+                self._schema_registry_client.register_schema
+                if self._auto_register_schemas
+                else self._schema_registry_client.get_schema_id
+            )
+            schema_id = schema_id_func(
+                self._schema_group, schema_name, "Avro", schema_str, **kwargs
             ).schema_id
```

Comment:
this is smart, I would go with moving the logic to the `__init__` so that when serialization/deserialization, the if/else check doesn't get executed each time, which might be a performance overhead.